bitshuffle.py

```diff
@@ -137,21 +143,48 @@ def main():
 
     args = parser.parse_args()
 
+    # Checks if no parameters were passed so the help can be printed
+    if len(sys.argv[1:]) == 0:
+        parser.print_help()
+        exit(0)
+
+    if args.version:
+        print("Version: bitshuffle v{0}".format(program_version))
+        exit(0)
+
     if args.filename is None:
         args.filename = os.path.basename(args.input)
 
-    if args.encode:
-        with open(args.input, 'rb') as f:
-            packets = encode_file(f, args.chunksize, args.compresslevel,
-                                  args.filename)
-            with open(args.output, 'w') as of:
-                for p in packets:
-                    of.write(p)
-                    of.write("\n\n")
+    # Encode & Decode smart inference
+    if not args.encode and not args.decode:
+        # Assume Decode
+        if args.output and args.input == '/dev/stdin':
+            if stdin.isatty():
```

Comment:
I am not convinced this is an accurate heuristic. I would like bitshuffle to support piping, so that something like `curl <url> | bitshuffle` would decode a message. Perhaps we could reverse this, so that if args.input is *not* /dev/stdin decoding is the default, and encoding otherwise.

Reply:
@jyn514 Okay no a problem. So, would you like piping to default to decoding?

Reply:
I would prefer so. @charlesdaniels , do you have an opinion?